
///////////////////////////////////////////////////////////////////////////////
//
//  Copyright (c) 2008, Arizona State University
//  All rights reserved.
//  BSD License: http://www.opensource.org/licenses/bsd-license.html
//  Author: Jeff Conner
//
///////////////////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////////////////
//
//  Input dialog for files.
//
///////////////////////////////////////////////////////////////////////////////

#include "NewVaporIntrusion.h"

#include "Usul/Convert/Convert.h"

///////////////////////////////////////////////////////////////////////////////
//
//  Constructor.
//
///////////////////////////////////////////////////////////////////////////////

NewVaporIntrusion::NewVaporIntrusion ( QWidget *parent ) : BaseClass ( parent )
{
  // Initialize code from Designer.
  this->setupUi ( this );

 
}


///////////////////////////////////////////////////////////////////////////////
//
//  Destructor.
//
///////////////////////////////////////////////////////////////////////////////

NewVaporIntrusion::~NewVaporIntrusion()
{

}


///////////////////////////////////////////////////////////////////////////////
//
//  Get the expirement space name
//
///////////////////////////////////////////////////////////////////////////////

std::string NewVaporIntrusion::name()
{
  std::string n ( spaceName->text().toStdString() );

  return n;
}


///////////////////////////////////////////////////////////////////////////////
//
//  Get the width value (x)
//
///////////////////////////////////////////////////////////////////////////////

unsigned int NewVaporIntrusion::x()
{
  std::string value ( xValue->text().toStdString() );

  // x value
  unsigned int x ( 5 );

  if( value.length() > 0 )
  {
    x = Usul::Convert::Type< std::string, unsigned int >::convert( value );
  }

  return x;
}


///////////////////////////////////////////////////////////////////////////////
//
//  Get the height value (y)
//
///////////////////////////////////////////////////////////////////////////////

unsigned int NewVaporIntrusion::y()
{
  std::string value ( yValue->text().toStdString() );

  // y value
  unsigned int y ( 5 );

  if( value.length() > 0 )
  {
    y = Usul::Convert::Type< std::string, unsigned int >::convert( value );
  }

  return y;

}


///////////////////////////////////////////////////////////////////////////////
//
//  Get the depth value (z)
//
///////////////////////////////////////////////////////////////////////////////

unsigned int NewVaporIntrusion::z()
{
  std::string value ( zValue->text().toStdString() );

  // z value
  unsigned int z ( 5 );

  if( value.length() > 0 )
  {
    z = Usul::Convert::Type< std::string, unsigned int >::convert( value );
  }

  return z;

}